### ISSUE TITLE
InWorldz

### ISSUE THREAD

**Issue by SaraNyn** [AVsitter/AVsitter#48]
The further InWorldz testing has progressed the more issues have popped up and been rectified. As of now everything seems to be working except single sitter poses which really don't do much of anything at all **unless i add the lines
ANIM_INDEX = index;
integer broadcast = TRUE;
send_anim_info(broadcast)**
at line 264 in [AV]sitB.
Now InWorldz has a different way of implementing of implementing avatar as a prim as explained by my dearest friend Christine Nyn; _"Originally the avatar was recorded as being the last link but could not be manipulated like a link. When IW was upgraded to AAAP backwards compatible behaviour was retained and AAAP would only kick in if code attempting to move the avatar as a prim was detected. After that point AAAP would be in force."_
The upshot of all this is I'm here seeking advice as to how to get past this final hurdle (correctly as opposed to my jury rig which has minor adverse effects) in order to get this thing released in InWorldz at long last

**Comment by Sei-Lisa** [comment]
Hi SaraNyn,

Unfortunately we have no developers working on Halcyon support; only on SL and OpenSim 0.8.2+. Unless there is some other developer willing to get an InWorldz account and investigate, I'm afraid the research necessary to get this working is up to you or anyone with good knowledge of the Halcyon engine specifics, willing to spend time on AVsitter to find out how to properly add support for Halcyon in it.

I wouldn't discard the possibility that, being such a popular product in SL, InWorldz maintainers take interest in getting AVsitter to work in Halcyon. Perhaps reporting this incompatibility to them is a good idea.

If someone is willing to contribute code and get it to work on Halcyon, pull requests are welcome. The requirements are:

- Adding support for Halcyon (InWorldz) should not be a drag to the SL version.
  - There is a filter that detects specially formatted comments to activate OpenSim-only code, so that code specific to the OpenSim version does not affect the SL version. That filter is used by the OpenSim release generation makefile. A similar provision can be used for Halcyon, if it doesn't have a big impact on readability and maintainability.
- Contributor guidelines in https://avsitter.github.io/contribute.html should be followed.
- There are some additional contributor guidelines pending to be added to these, and they should be followed as well: https://github.com/AVsitter/avsitter.github.io/issues/15
- While not essential, a commitment to update Halcyon support as changes are made to the engine would be greatly appreciated, lest it becomes outdated and forgotten.

For the record, inserting the lines you quote in [line 269 of the current version of [AV]sitB](https://github.com/AVsitter/AVsitter/blob/032e0369bf25ae36a47c9a591b4bea3efa3a8b89/AVsitter2/%5BAV%5DsitB.lsl#L269) would make no sense and make the code misbehave. That's why it's important to submit pull requests. Pointing to a line without specifying the commit ID it belongs to isn't helpful. Pull requests reference a specific version at the moment the modification is created, and can most of the times be adapted to the most recent version with minimal effort if any. They have the additional benefit that credit for that commit goes to its author, and they make the life easier to the maintainers who don't have to look up what version to apply the patch to. For these reasons, and as specified in the guidelines, pull requests are the way to submit patches.

**Comment by SaraNyn** [comment]
What should I do when I solve it, which I believe I just did?

**Comment by SaraNyn** [comment]
I've attached the a zipped version of the Core InWorldz scripts and will explain the differences if anyone can then point me towards how to make all that into a pull request I will do so. 
[AV]sitB uses "string SEP;" at line 51 and comments out "if (id == MY_SITTER)" at line 400. [AV]sitA uses "string SEP;" at line 85. [AV]object comments out matters related to SL experiences beginning at line 144 through line 194 and then line 221.
Those are the scripting differences from the main branch.

[AVsitter 2.2 InWorldz.zip](https://github.com/AVsitter/AVsitter/files/1339575/AVsitter.2.2.InWorldz.zip)

**Comment by Sei-Lisa** [comment]
Thank you. We use the "fork and pull" collaboration model. To set you up so you can start collaborating with a project, you fork that project, using the Fork button in the top right. That basically creates a clone of the main repository in your GitHub account, but it can evolve independently of the main repository. You crate branches in order to work on changes that you want to announce to the maintainers of the main repository, then make changes by committing to these branches, and then you create a pull request out of the branch in question.

Check GitHub help, especially these pages:

https://help.github.com/categories/collaborating-with-issues-and-pull-requests/
https://help.github.com/articles/proposing-changes-to-your-work-with-pull-requests/

It helps a lot to be well acquainted with Git when you read these GitHub-specific pages, so if you want to learn about using Git first, here's a page that may be of help:

https://help.github.com/categories/using-git/

**Comment by badwolf5213** [comment]
I downloaded @SaraNyn Halcyon version and uploaded it into my Halcyon grid (based on current master branch as of 05/08/21) and found that it did not work. 
Did some exploring of the scripts and found in AVadjuster.lsl that 3 instances of "mainscript" needed to be changed to "main_script".
I forked @SaraNyn  version, published it to my github (https://github.com/badwolf5213/AVsitter) and also created a pull request on @SaraNyn  fork. Will continue to look for any other issues as I build and update my github as needed.

**Comment by SaraNyn** [comment]
The last version I posted, which I'm not sure I see here, worked well enough (Inworldz) we used it in several commercial products (Which sold pretty well) and never received a single complaint........
Halcyon *has* been updated since then so that may explain the reason for changes being needed. I hope the version I did was helpful in any case.

**Comment by badwolf5213** [comment]
@SaraNyn YES!!! Your code was invaluable! It allowed us to compare @Sei-Lisa current code with your code and find the fix.

Yes, more than likely there has been a server side change that necessitated the change. You have more than likely been spending your time on an OpenSim based grid or SL and didn't catch the change. Very understandable.

Anyhoo, we caught the change, made the fix and you are MORE THAN WELCOME to incorporate it. I believe I did do a pull request on your branch.

**Comment by SaraNyn** [comment]
InWorldz hasn't existed for over four years is why I've been inactive
